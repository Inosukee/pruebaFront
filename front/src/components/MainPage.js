import React from 'react'
import { History } from './History'
import { ProductsPage } from './ProductsPage'


export const MainPage = () => {
    return (
        <div className='Main__Main'>
            <History />
            <ProductsPage />
        </div>
    )
}
