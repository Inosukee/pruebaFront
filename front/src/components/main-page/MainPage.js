import React from 'react'
import { useFetchProduct } from '../../hooks/useFetch';
import { History } from '../history/History'
import { ProductsPage } from '../products/ProductsPage'


export const MainPage = ({ filter }) => {

    const { data: products, loading } = useFetchProduct(filter);
    return (
        <>
            {loading && <p>Cargando...</p>}
            {
                products.length > 0 &&
                <div className='main__main animate__animated animate__fadeIn'>
                    <History />
                    <ProductsPage productos={products} />
                </div>
            }
        </>
    )
}
