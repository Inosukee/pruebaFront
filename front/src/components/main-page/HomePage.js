import React from 'react'
import { MainPage } from '../main-page/MainPage'
import { Header } from '../shared/Header'


export const HomePage = () => {
    return (
        <>
            <Header />
            <MainPage />
        </>
    )
}
