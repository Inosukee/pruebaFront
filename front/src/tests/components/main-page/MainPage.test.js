import React from 'react'
import { shallow } from 'enzyme';
import { MainPage } from '../../../components/main-page/MainPage';


describe('Pruebas en <MainPage />', () => {
  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<MainPage filter={'Comida'} />)
    expect(wrapper.find('.main__main').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})