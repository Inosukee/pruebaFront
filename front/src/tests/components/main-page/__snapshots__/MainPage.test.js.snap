// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MainPage /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <p>
    Cargando...
  </p>
  <div
    className="main__main animate__animated animate__fadeIn"
  >
    <History />
    <ProductsPage />
  </div>
</Fragment>
`;
